src/Domain/LeanCode.DomainModels.MassTransitRelay/Middleware/StoreAndPublishEventsFilter.cs

```diff
@@ -24,12 +24,15 @@ public StoreAndPublishEventsFilter(AsyncEventsInterceptor interceptor, EventsSto
     {
         var events = await interceptor.CaptureEventsOfAsync(() => next.Send(context));
 
-        await store.StoreAndPublishEventsAsync(
-            events,
-            context.ConversationId,
-            new EventPublisher(context),
-            context.CancellationToken
-        );
+        if (context.GetRedeliveryCount() == 0)
+        {
+            await store.StoreAndPublishEventsAsync(
```

Comment:
This means that any event raised during redelivery wouldn't be published. Also if the consumer relies on this piece to save it's data - it'd also be lost.